### PR TITLE
Add inventory

### PR DESCRIPTION
**Pull request by joon-ko**
Adds the inventory as a autoload. Also added an enum for ingredients! If we think the inventory will have any non-ingredients, we can probably rename the Ingredient enum to something more generic.

### CODE REVIEW

File: godot/Ingredient.gd [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class_name Ingredient
```
Comment (alexmat2on) [inline godot/Ingredient.gd:1]: I'd probably just make this more generic and call it `Item` but im okay with doing that in a later change :shrug:

**Review by alexmat2on — APPROVED** [review]
:tada:

File: godot/Ingredient.gd (hunk 1 above)
Comment (alexmat2on) [reply godot/Ingredient.gd:1]: bleh I meant in the Inventory node, this is fine since its just the ingredients in general

Comment (joon-ko) [reply godot/Ingredient.gd:1]: yeah i was debating whether to use item or ingredient im ok with either! ill just push this up for now :p

**Pull request merged**